src/libAtomVM/module.c

```diff
@@ -312,12 +312,30 @@ Module *module_new_from_iff_binary(GlobalContext *global, const void *iff_binary
 COLD_FUNC void module_destroy(Module *module)
 {
     free(module->labels);
+    for (int i = 0; i < module->functions_count; ++i) {
+        const struct ExportedFunction *fun = module->imported_funcs[i];
+        switch (fun->type) {
+            // Preallocated function types
+            case NIFFunctionType:
+            case BIFFunctionType:
+            case GCBIFFunctionType:
+                break;
+            default:
+                free((void *) fun);
+        }
+    }
     free(module->imported_funcs);
     free(module->literals_table);
     free(module->local_atoms_to_global_table);
     if (module->free_literals_data) {
         free(module->literals_data);
     }
+    free(module->line_refs);
+    free(module->filenames);
```

Comment:
I think this makes sense only in `main` since `filenames` is not available in release-0.6 branch.
The correct fix would be splitting this PR in 2, one for release-0.6 and one for main, _but_:
this memory leak for now is relevant only during tests so it doesn't have any practical impact on devices running release-0.6, so we can go for the lazy option that it pointing to main branch.